src/Module/Moderation/Report/Create.php

```diff
@@ -0,0 +1,325 @@
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->systemMessages = $systemMessages;
+		$this->page           = $page;
+		$this->session        = $session;
+		$this->factory        = $factory;
+		$this->repository     = $repository;
+	}
+
+	protected function post(array $request = [])
+	{
+		if (!$this->session->getLocalUserId()) {
+			throw new ForbiddenException();
+		}
+
+		$report = [];
+		foreach (['cid', 'category', 'rule-ids', 'uri-ids'] as $key) {
+			if (isset($request[$key])) {
+				$report[$key] = $request[$key];
+			}
+		}
+
+		if (isset($request['url'])) {
+			$cid = Contact::getIdForURL($request['url']);
+			if ($cid) {
+				$report['cid'] = $cid;
+			} else {
+				$report['url'] = $request['url'];
+				$this->systemMessages->addNotice($this->t('Contact not found or their server is already blocked on this node.'));
+			}
+		}
+
+		if (isset($request['comment'])) {
+			$this->session->set('report_comment', $request['comment']);
+			unset($request['comment']);
+		}
+
+		if (isset($request['report_create'])) {
+			$report = $this->factory->createFromForm(
+				System::getRules(true),
+				$request['cid'],
+				$this->session->getLocalUserId(),
+				$request['category'],
+				!empty($request['rule-ids']) ? explode(',', $request['rule-ids']) : [],
+				$this->session->get('report_comment') ?? '',
+				!empty($request['uri-ids']) ? explode(',', $request['uri-ids']) : [],
+			);
+			$this->repository->save($report);
+		}
+
+		$this->baseUrl->redirect($this->args->getCommand() . '?' . http_build_query($report));
+	}
+
+	protected function content(array $request = []): string
+	{
+		if (!$this->session->getLocalUserId()) {
+			throw new ForbiddenException($this->t('Please login to access this page.'));
+		}
+
+		$this->page['aside'] = $this->getAside($request);
+
+		if (empty($request['cid'])) {
+			return $this->pickContact($request);
+		}
+
+		if (empty($request['category'])) {
+			return $this->pickCategory($request);
+		}
+
+		if ($request['category'] == Report::CATEGORY_VIOLATION && !isset($request['rule-ids'])) {
+			return $this->pickRules($request);
+		}
+
+		if (!isset($request['uri-ids'])) {
+			return $this->pickPosts($request);
+		}
+
+		return $this->summary($request);
+	}
+
+	private function pickContact(array $request): string
+	{
+		$tpl = Renderer::getMarkupTemplate('moderation/report/create/pick_contact.tpl');
+		return Renderer::replaceMacros($tpl, [
+			'$l10n' => [
+				'title'       => $this->t('Create Moderation Report'),
+				'page'        => $this->t('Pick Contact'),
+				'description' => $this->t('<p>Please enter below the contact address or profile URL you would like to create a moderation report about.</p>'),
```

Comment:
Could you please move the HTML tags outside of the translation-able string? I think in this case it is not needed inside of the translation.